routing/RouteSolver.py

```diff
@@ -0,0 +1,134 @@
+import sys
+import re
+##################################################################################
+##                                                                              ##
+##                            Global Variables                                  ##
+##                                                                              ##
+##################################################################################
+graph = Graph.Graph()
+##################################################################################
+##                                                                              ##
+##                            Functions/Methods                                 ##
+##                                                                              ##
+##################################################################################
+
+def is_int(s):
+    try:
+        int(s)
+        return True
+    except ValueError:
+        return False
+
+def loadConfig(configFile):
+    f = open(configFile,"r")
+    lines = f.readlines()
+    currNode = ""
+    bad_chars = '(){}<>[]'
+    for i in range(0,len(lines)):
+        line= lines[i]
+        if "Next Node:" in line:
+            graph.add_vertex(lines[i+1].strip())
+            currNode = lines[i+1].strip()
+        elif "Connected Nodes:" in line:
+            connectedNodes = lines[i+1].split(';')
+            for node in connectedNodes:
+                nodeName = node.split(',')
+                for c in bad_chars: nodeName[0] = nodeName[0].replace(c, "")
+                for c in bad_chars: nodeName[1] = nodeName[1].replace(c, "")
+                nodeName[1].rstrip()
+                #graph.add_edge((currNode,(nodeName[0],nodeName[1])))
+                # Adding directions is causing an issue. May have to come up with
+                #another solution for direction
+                graph.add_edge((currNode,nodeName[0]))
+
+def getGraphDetailsFromUser():
+    numberOfNodes = 0;
+    while True:
+        numberOfNodes = input("How many nodes are in your graph? ")
+        if(is_int(numberOfNodes)):
+            break
+    nodeCount = 0
+    while (nodeCount < int(numberOfNodes)):
+        newName = input("Please enter a name for node {}\n".format(nodeCount))
+        graph.add_vertex(newName)
+        nodeCount += 1
+    print(graph.vertices())
+    for node in graph.vertices():
+        anotherNode = True
+        while anotherNode:
+            nodeName = input("Please enter connected node name for node {}:\n".format(node))
+            if(nodeName in graph.vertices()):
+                while True:
+                    direction = input("Please enter the cardinal direction traveling from {} to {}: (N E S W)\n".format( node, nodeName))
+                    if direction is "N" or "S" or "E" or "W":
+                        graph.add_edge((node,nodeName))
+                        anotherConnection = input("Is there another connected node? (y/n)\n")
```

Comment:
Refactor you `[y/n]` input prompts as follows

```python
def get_bool_input(message):
    '''Gets y/n input for given message'''
    while True:
        ans = input(f'{message} (y/n): ')
        if ans.lower() == 'y':
            return True
        elif ans.lower() == 'n':
            return False
        else:
            print(f'Invalid input: {ans} - y/n or Y/N only')
            continue
```

That should simplify the logic